ilc/server/app.js

```diff
@@ -24,6 +27,15 @@ module.exports = (registryService, pluginManager) => {
         await i18nOnRequest(req, reply);
     });
 
+    if (transitionHooksPlugin !== null) {
+        app.addHook('preHandler', async (req, reply) => {
```

Comment:
Let's discuss... Changes from my PR may help to do this part of logic in a better way.

Reply:
Updated